File: composite/en_us/cpp/sources/user_story.cpp
#include "user_story.h"

UserStory::UserStory(string title, string role, string goal, string reason)
{
	/*
	 * Assign the title, if exists.
	 */
	if(!title.empty())
	{
		m_title.assign(title);
	}
	else
	{
		m_title.assign(NOT_DEFINED);
	}

	/*
	 * Assign the role, if exists.
	 */
	if(!role.empty())
	{
		m_role.assign(role);
	}
	else
	{
		m_role.assign(NOT_DEFINED);
	}

	/*
	 * Assign the goal, if exists.
	 */
	if(!goal.empty())
	{
		m_goal.assign(goal);
	}
	else
	{
		m_goal.assign(NOT_DEFINED);
	}

	/*
	 * Assign the reason, if exists.
	 */
	if(!reason.empty())
	{
		m_reason.assign(reason);
	}
	else
	{
		m_reason.assign(NOT_DEFINED);
	}
}

void
UserStory::show_requirement()
{
	/*
	 *	Show the User Story's information.
	 */
	cout
		<< "[USER STORY]: " << m_title << endl
		<< "\t" << "As: " << m_role << endl
		<< "\t" << "I want to: " << m_goal << endl
		<< "\t" << "So that: " << m_reason <<
	endl;
}

File: composite/en_us/java/UserStory.java
/**
 * File: UserStory
 * Role: Leaf
 * Purpose: Specif requirement type
 */
public class UserStory extends Requirement{

  private String role;
  private String goal;
  private String reason;

  public UserStory (String title, String content, String role, String goal, String reason) {
    super(title, content);
    this.role = role;
    this.goal = goal;
    this.reason = reason;
  }

  @Override
  public String showRequirement(){
    return "[User Story]: " + getTitle() +
            "\nAs " + getRole() +
            "\nI want to " + getGoal() +
            "\nSo that " + getReason();
  }

  public void setRole (String role){
    this.role = role ;
  }

  public String getRole() {
    return role;
  }

  public void setGoal (String goal){
    this.goal = goal;
  }

  public String getGoal() {
    return goal;
  }

  public void setReason (String reason){
    this.reason = reason ;
  }

  public String getReason() {
    return reason;
  }

}
